lexer/lexer.hpp

```diff
@@ -1,12 +1,45 @@
+#pragma once
+
+#include <functional>
+#include <cctype>
 #include "trie.hpp"
 #include "token.hpp"
-#include "trie.hpp"
-#pragma once
 
-namespace lexer
-{
-    std::vector<Token> read(std::string);
-    common::Trie<TokenType> getTrie();
-    std::vector<Token> splitLine(std::string);
-    void freeResult();
-} // namespace lexer
\ No newline at end of file
+namespace lexer {
+
+
+extern common::Trie<TokenType> g_keywordTrie;
+extern common::Trie<TokenType> g_operatorTrie;
+
+class Lexer {
+public:
+    Lexer(std::string src) : m_buf(src) {}
+
+    Token Next();
+
+private:
+    std::string m_buf;
+    size_t m_pos = 0, m_lastAfterNewLine = 0;
+    size_t m_lineNum = 1;
+
+    static bool isspace(char ch) {
+        return ch != '\n' && std::isspace(ch);
+    }
+
+    static bool isdigit(char ch) {
+        return std::isdigit(ch);
```

Comment:
_(Just out of curiosity)_
Is it possible to write something like `constexpr auto isdigit = std::isdigit;`?
How about just `using std::isdigit;`?